### PR TITLE
Track account request

### PR DESCRIPTION
**Pull request by briankfitzwater**
- Adds sqlite database to remember InstanceID to RequestID mapping (uses GORM)
- last_operation performs DescribeCreateAccountStatus of RequestID
- Provision saves InstanceID and RequestID to database
- AccountName is set to InstanceID (to ensure name is unique)

### CODE REVIEW

**Comment by briankfitzwater** [comment]
[As a DevSecOps team member, I need brokered AWS accounts to be tracked](https://trello.com/c/dJjUxKxL/338-as-a-devsecops-team-member-i-need-brokered-aws-accounts-to-be-tracked)

**Comment by briankfitzwater** [comment]
[As a tenant, I want to know the status of my brokered AWS account](https://trello.com/c/vhwp1oj3/341-as-a-tenant-i-want-to-know-the-status-of-my-brokered-aws-account)

File: broker.go [hunk 1]
```diff
@@ -68,20 +71,29 @@ func (b awsAccountBroker) Services(ctx context.Context) []brokerapi.Service {
...
 	if err != nil {
 		return spec, err
 	}
-	// TODO use the result?
 
 	b.logger.Info("Account created for " + email)
+	var requestID = *createResult.CreateAccountStatus.Id
+	b.logger.Info("RequestId: " + requestID)
+
+	b.db.Create(&serviceInstance{InstanceId: instanceID, RequestId: requestID})
```
Comment (afeld) [inline broker.go]: Minor: I’d refactor the persistence out of Provision.

Comment (briankfitzwater) [reply broker.go]: I agree...just not sure where to put it.

**Comment by briankfitzwater** [comment]
Instead of installing sqlite to setup the database schema, I used GORM...

```golang
package main

import (
  "fmt"
	"github.com/jinzhu/gorm"
	_ "github.com/jinzhu/gorm/dialects/sqlite"
)

type serviceInstance struct {
  gorm.Model
  InstanceId string
  RequestId string
}

func main() {
  db, err := gorm.Open("sqlite3", "aws-account-broker.db")
  if err != nil {
    panic("failed to connect database")
  }
  defer db.Close()

  // Migrate the schema
  db.AutoMigrate(&serviceInstance{})

  // Create
  db.Create(&serviceInstance{InstanceId: "scp-test", RequestId: "car-2071c37006f911e89f98500c04b7b2c5"})
  db.Create(&serviceInstance{InstanceId: "broker-account", RequestId: "car-9f57a600103611e88d8f50843ec7f68d"})

  //// Read
  var instance serviceInstance
  db.First(&instance, "instance_id = ?", "scp-test")
  fmt.Println(instance.RequestId)
}
```

I really didn't want to add that to the README.

**Comment by briankfitzwater** [comment]
@afeld Shouldn't the package name be `aws-account-broker` instead of `main`

**Comment by briankfitzwater** [comment]
These are working e-mails.  At least for Google...see [article](https://gmail.googleblog.com/2008/03/2-hidden-ways-to-get-more-from-your.html)

**Comment by Vermyndax** [comment]
@briankfitzwater Just because it works in public Gmail does not guarantee that it will work in GSA's GMail Suite, or is that incorrect?

I know that in the case of other folks, they will not be able to work with this schema.

**Comment by Vermyndax** [comment]
Just to be clear, I'm not going to block this PR for the email changes, as I've filed an issue on that. It should be fixed in another branch. Guess I should scope out that review feedback ;)

**Comment by afeld** [comment]
--> #2 for the email convo.

@briankfitzwater Let us know when you'd like a re-review!

> Shouldn't the package name be `aws-account-broker` instead of `main`

No; "Executable commands must always use package `main`."

https://golang.org/doc/code.html#PackageNames

**Comment by briankfitzwater** [comment]
@afeld Tests are working

File: README.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ This is an API that [creates AWS (sub)accounts in an Organization](https://docs.
 1. Install system dependencies.
     1. [Go](https://golang.org/)
     1. [Dep](https://golang.github.io/dep/docs/installation.html)
+    1. [SQLite](https://www.sqlite.orga) - Proof of concept testing is using SQLite3 for persistence
+        -
```
Comment (afeld) [inline README.md]: Minor: errant `-` here that is doing weird things to the formatting.

File: broker_test.go [hunk 3]
```diff
@@ -19,8 +23,10 @@ type mockOrganizationsClient struct {
 }
 
 func (m mockOrganizationsClient) CreateAccount(input *organizations.CreateAccountInput) (*organizations.CreateAccountOutput, error) {
+	var id = "car-999999999999"
```
Comment (afeld) [inline broker_test.go]: Nitpick: common practice in Go is to use ["short assignment"](https://tour.golang.org/basics/10) for variables like this. In other words, this line would change to

```go
id := "car-999999999999"
```

Ditto for the other `var` usage in this file.

File: broker_test.go [hunk 4]
```diff
@@ -39,7 +60,18 @@ func mockBroker(createErr error, createState string) awsAccountBroker {
 	baseEmail := "foo@bar.com"
 	logger := lager.NewLogger("test")
 
-	return awsAccountBroker{mgr, baseEmail, logger}
+	//Make sure there is no existing DB
+	var dbFileName = filepath.Join(os.TempDir(), "broker_test.db")
+	//os.Remove(dbFileName)
```
Comment (afeld) [inline broker_test.go]: Did you mean to put this back in?

File: broker_test.go [hunk 5]
```diff
@@ -101,7 +133,16 @@ func TestProvisionSync(t *testing.T) {
...
+	_, err := broker.Provision(ctx, "test-account", details, false)
 
 	assert.Error(t, err)
 }
+
+func TestLastOperation(t *testing.T) {
+	broker := mockBroker(nil, organizations.CreateAccountStateSucceeded)
+	ctx := context.Background()
+
+	result, _ := broker.LastOperation(ctx, "test-account", "")
```
Comment (afeld) [inline broker_test.go]: I'd check that there was no error.

```go
assert.NoError(t, err)
```

**Comment by afeld** [comment]
Just the one little remaining thing. Feel free to merge after that!

**Pull request merged**